backend/services/implementations/userService.ts

```diff
@@ -143,6 +179,33 @@ class UserService implements IUserService {
     return userDtos;
   }
 
+  async getUserCountByTown(
+    startDate?: string,
+    endDate?: string,
+  ): Promise<{ [key: string]: number }> {
+    const groupByQuery = { $group: { _id: "$town", userCount: { $sum: 1 } } };
+
+    let query: Array<UserCountGroupByTown>;
+
+    if (!startDate && !endDate) {
+      query = await MgUser.aggregate([groupByQuery]).exec();
+    } else {
+      query = await MgUser.aggregate([
+        buildMatchObject(startDate, endDate),
+        groupByQuery,
+      ]).exec();
+    }
+
+    return query.reduce(
+      (map: { [key: string]: number }, obj: UserCountGroupByTown) => {
+        // eslint-disable-next-line no-underscore-dangle
+        map[obj._id] = obj.userCount;
```

Comment:
Wasn't sure how to handle this lint error. (Is this actually indicative that something is wrong? I don't see the reassignment)

Can disable by changing line 201 to 
`// eslint-disable-next-line no-param-reassign, no-underscore-dangle`

Reply:
Seems like this isn't a real error, and using that disable comment is fine. Seems like most people disable or even create a linter rule to ignore this error https://stackoverflow.com/a/66902167/10049850